scylla-cql/src/frame/value.rs

```diff
@@ -837,6 +838,70 @@ impl<T: ValueList> BatchValues for Vec<T> {
     }
 }
 
+/// Implements `BatchValues` from an `IntoIterator` over things that implement `ValueList`
+///
+/// This is to avoid requiring allocating a new `Vec` containing all the `ValueList`s directly:
+/// with this, one can write:
+/// ```ignore
+/// session.batch(&batch, BatchValuesFromIter::new(lines_to_insert.iter().map(|l| &l.value_list)))
+/// ```
+/// where `lines_to_insert` may also contain e.g. data to pick the statement...
+///
+/// The underlying iterator will always be cloned at least once, once to compute the length if it can't be known
+/// in advance, and be re-cloned at every retry.
```

Comment:
Is this true though or is it only serialized once?
Need to check this...

Reply:
Yes it is true, as the `SerializedRequest` is only built in `Connection::send_request`:
https://github.com/scylladb/scylla-rust-driver/blob/014c83a9d55d3beb9ac16a01ddec12cecaabe48c/scylla/src/transport/connection.rs#L684